### PR TITLE
Added precision option to round floating point values.

### PR DESCRIPTION
**Pull request by Digicrat**
IEEE 754 floating point numbers are not always precise, and rounding errors can accumulate when round-tripping data making comparisons difficult.  This is particularly common on certain platforms, such as cJSON, where conversions between float and double can aggravate matters.  

This change introduces a "--precision" option that will explicitly round all floating point numbers in the JSON file to the specified precision prior to comparison.

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -17,7 +18,8 @@ module.exports = function (argv) {
       '  -f, --full              Include the equal sections of the document, not just the deltas #var(full)',
       '  -o, --output-keys KEYS  Always print this comma separated keys, with their value, if they are part of an object with any diff #var(outputKeys)',
       '  -s, --sort              Sort primitive values in arrays before comparing #var(sort)',
-      '  -k, --keys-only         Compare only the keys, ignore the differences in values #var(keysOnly)'
+      '  -k, --keys-only         Compare only the keys, ignore the differences in values #var(keysOnly)',
+      '  -p, --precision PREC    Round all floating point numbers to this precision prior to comparison'
```
Comment (ewoudenberg) [inline lib/cli.js:22]: How about 
```
' -p, --precision PREC Round all floating point numbers to this number of decimal places prior to comparison'
```

File: lib/cli.js [hunk 2]
```diff
@@ -37,15 +39,21 @@ module.exports = function (argv) {
...
     process.stderr.write('Parsing new file...\n')
   }
-  const json2 = JSON.parse(data2)
+  json2 = JSON.parse(data2)
 
   if (options.verbose) {
     process.stderr.write('Running diff...\n')
   }
+
+  if (options.precision) {
```
Comment (ewoudenberg) [inline lib/cli.js:52]: How about:
```
if (options.precision != undefined) {
```
In this way, a precision of 0 can be specified.

**Review by ewoudenberg — CHANGES_REQUESTED** [review]
Thank you for your submission! If you can address the comments and adjust the unit tests so that:

1. All existing tests pass
2. There are new test(s) for the precision feature.

I will be happy to merge your change.

**Comment by Digicrat** [comment]
I fixed the existing test and made the suggested changes.

I wasn't able to add a new unit test though, as the way the existing tests are written the new logic in cli.js isn't invoked.

**Comment by ewoudenberg** [comment]
Thank you! I've altered the PR slightly and merged it.
https://github.com/andreyvit/json-diff/pull/75

**Pull request closed without merging**